sdk/core/azure-core/src/http/transport_policy.cpp

```diff
@@ -48,16 +48,16 @@ std::unique_ptr<RawResponse> TransportPolicy::Send(
   auto statusCode = static_cast<typename std::underlying_type<Http::HttpStatusCode>::type>(
       response->GetStatusCode());
 
-  if (request.IsDownloadViaStream() && statusCode < 300)
+  if (!request.IsBufferedDownload() && statusCode < 300)
   { // special case to return a response with BodyStream to read directly from socket
     // Return only if response is valid (less than 300)
     return response;
   }
 
   // default behavior for all request is to download body content to Response
   // If ReadToEnd fail, retry policy will eventually call this again
-  // Using DownloadViaStream and getting an error code would also get to here to download error from
-  // body
+  // Using `bufferedDownload=false` and getting an error code would also download error to the
+  // body buffer
```

Comment:
I don't follow what this comment is saying.
1) Where are we setting bufferedDownload=false in this code path/function?
2) What does it mean to `download error to the body buffer`?

Reply:
I also don't follow. BTW, did you mean to say `bufferedDownload==false` (i.e. `==` instead of `=`). Not that it would make things clearer for me.

Reply:
Updated comments. Let me know if that's better now.

Reply:
LGTM!